contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3Propagator.java

```diff
@@ -51,19 +51,31 @@
   private final B3PropagatorInjector b3PropagatorInjector;
   private final B3PropagatorExtractor b3PropagatorExtractor;
 
-  /** Creates a new instance of {@link B3Propagator}. Defaults to use multiple headers. */
-  public B3Propagator() {
-    this(false);
-  }
+  /**
+   * Creates a new instance of {@link B3Propagator}. This instance contains the reference to {@link
+   * B3PropagatorInjectorSingleHeader} and {@link B3PropagatorExtractorSingleHeader}.
+   */
+  public static final B3Propagator SINGLE_HEADER = new B3Propagator(/* singleHeader= */ true);
+
+  /**
+   * Creates a new instance of {@link B3Propagator}. This instance contains the reference to {@link
+   * B3PropagatorInjectorMultipleHeaders} and {@link B3PropagatorExtractorMultipleHeaders}.
+   */
+  public static final B3Propagator MULTI_HEADER = new B3Propagator(/* singleHeader= */ false);
 
   /**
    * Creates a new instance of {@link B3Propagator}.
    *
    * @param singleHeader whether to use single or multiple headers.
    */
-  public B3Propagator(boolean singleHeader) {
-    b3PropagatorInjector = new B3PropagatorInjector(singleHeader);
-    b3PropagatorExtractor = new B3PropagatorExtractor(singleHeader);
+  private B3Propagator(boolean singleHeader) {
```

Comment:
since this constructor is private, and only called from the field declarations, you could just inline the boolean logic into the call sites (i.e. get rid of the boolean and just pass in the two implementations that you want in the right constructor calls)

Reply:
Took care of it.